### DISCUSSION TITLE
Why i use ffmpeg push srt stream show a lot /ffmpeg*E:SRT.d: SND-DROPPED 30 packets - lost delaying for 1049ms

### DISCUSSION THREAD

**Discussion by xiaolitongxue666** [Haivision/srt#2755] [Q&A] [answered]
My env : 

OS : Ubuntu 20.04.6 LTS x86_64
Kernel: 5.15.0-75-generic
CPU : CPU: AMD Ryzen 5 5600X (12) @ 3.700GHz
Memory: 465MiB / 15893MiB

FFMpeg : ffmpeg version 4.4 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.4.0-1ubuntu1~20.04.1)

FFmpeg test command : 
```
ffmpeg -v "debug" -re -stream_loop -1 -i sample_1280x720_surfing_with_audio2.ts -max_muxing_queue_size 1024 -c copy -strict -2 -y -f mpegts 'srt://live-push.XXXXXX.com:1937?streamid=#!::h=live-push.XXXXXXX.com,r=live-bvc/?streamname=live_XXXXXXXXXXXXXXX,key=233a9627b9XXXXXXXXXXXXXX2a,schedule=srtts,pflag=1/?streamname=live_XXXXXXXXXXXXXX&key=23XXXXXXXXXXXXXX0bef2a'
```
When i use this video as input , push srt work well
```
# ffprobe info
Input #0, mpegts, from 'sample_1280x720_surfing_with_audio2.ts':
  Duration: 00:03:03.08, start: 1.462089, bitrate: 1916 kb/s
  Program 1 
    Metadata:
      service_name    : Service01
      service_provider: FFmpeg
  Stream #0:0[0x100]: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(progressive), 1280x720 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 90k tbn, 47.95 tbc
  Stream #0:1[0x101](eng): Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp, 132 kb/s
```

But this one , push srt show a lot blow : 

```
# ffprobe info
[h264 @ 0x559d1b9616c0] non-existing PPS 0 referenced
    Last message repeated 1 times
[h264 @ 0x559d1b9616c0] decode_slice_header error
[h264 @ 0x559d1b9616c0] no frame!
Input #0, mpegts, from '1080P60AVC_MPEG1.ts':
  Duration: 00:09:59.05, start: 206.201278, bitrate: 7988 kb/s
  Program 2 
    Metadata:
      service_name    : prog2
      service_provider: 
  Stream #0:0[0x81]: Video: h264 (Main) ([27][0][0][0] / 0x001B), yuvj420p(pc, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 60 fps, 59.94 tbr, 90k tbn, 120 tbc
  Stream #0:1[0x91]: Audio: mp2 ([3][0][0][0] / 0x0003), 48000 Hz, stereo, fltp, 192 kb/s
```

```
/ffmpeg*E:SRT.d: SND-DROPPED 30 packets - lost delaying for 1049ms
```

**Comment by ethouris** [comment, marked as answer]
This message means that the sender decided to not retransmit the lost packet upon request because it won't reach the receiver before it's time to play the packet following it, which will cause dropping this one, so even if it is retransmitted, the receiver would still drop it.

This mechanism can be switched off (`snddropdelay` option, see `SRTO_SNDDROPDELAY` in [the documentation](../blob/master/docs/API/API-socket-options.md)). 

If dropping the packets is caused by largely fluctuating transmission speed cap, it could be mitigated by increasing the latency, which will give SRT more time to recover lost packets.

**Reply by xiaolitongxue666** [reply]
Thanks for you help, but i still want to ask how to  increasing the latency ?

**Reply by xiaolitongxue666** [reply]
Finally i solved this problem.
1 - Add snddropdelay option with value -1 
2 - Exchange my route to a better one 

At last this is my ffmpeg command : 
```
/tmp/ffmpeg -v "info" -re -stream_loop -1 -i 1080p60.mp4 -max_muxing_queue_size 1024 -acodec aac -vcodec libx264  -r 60.000000 -latency 10000 -rcvlatency 10000 -snddropdelay -1 -send_buffer_size 10000000 -recv_buffer_size 10000000 -f mpegts 'srt://live-push.XXXXXXX.com:1937?streamid=#!::h=live-push.XXXXXXX.com,r=live-bvc/?streamname=live_XXXXXXX_46330969,key=XXXXXXXef2a,schedule=srtts,pflag=1/?streamname=live_XXXXXXX_46330969&key=233a96XXXXXXXa10bef2a?' 

```

@ethouris Thanks a lot

**Comment by xiaolitongxue666** [comment]
> This message means that the sender decided to not retransmit the lost packet upon request because it won't reach the receiver before it's time to play the packet following it, which will cause dropping this one, so even if it is retransmitted, the receiver would still drop it.
> 
> This mechanism can be switched off (`snddropdelay` option, see `SRTO_SNDDROPDELAY` in [the documentation](../blob/master/docs/API/API-socket-options.md)).
> 
> If dropping the packets is caused by largely fluctuating transmission speed cap, it could be mitigated by increasing the latency, which will give SRT more time to recover lost packets.
